utils/include/edm4hep/utils/ParticleIDUtils.h

```diff
@@ -0,0 +1,96 @@
+#ifndef EDM4HEP_UTILS_PARTICLEIDUTILS_H
+#define EDM4HEP_UTILS_PARTICLEIDUTILS_H
+
+#include <edm4hep/ParticleIDCollection.h>
+#include <edm4hep/ReconstructedParticle.h>
+
+#include <podio/Frame.h>
+
+#include <map>
+#include <optional>
+#include <string>
+#include <type_traits>
+#include <vector>
+
+namespace edm4hep::utils {
+
+/// A simple struct bundling relevant metadata for a ParticleID collection
+struct ParticleIDMeta {
+  std::string algoName{};                ///< The name of the algorithm
+  int32_t algoType{0};                   ///< The (user defined) algorithm type
+  std::vector<std::string> paramNames{}; ///< The names of the parameters
+};
+
+/// Get the index of the parameter in the passed ParticleID meta info
+std::optional<int> getParamIndex(const ParticleIDMeta& pidMetaInfo, const std::string& param);
+
+/// Utility class to invert the ParticleID to ReconstructedParticle relation
```

Comment:
Could you add a code example how the PIDHandler is supposed to be used?
Created, filled, information extracted?

Reply:
Done. Visible [here](https://github.com/tmadlener/EDM4hep/blob/particleid-reversal/doc/PIDHandler.md)